docs: Complete Documentation Rewrite

cutterbl opened the pull request:
This is an exhaustive overhaul of documentation of Big Calendar. Here's the basic highlights:

- All documentation has been moved to Storybook
- Storybook has been upgraded
- All previous Examples have been updated and included
- Stories, with mdx docs and specific examples, have been created for every prop
- Addon stories, to include examples and prop stories, have been created for DnD
- Guide stories have been created to talk about Custom Styling, Custom Views, information about Big Calendar and Dates, and a guide on dealing with Timezones
- General documentation contribution guidelines have been included
- A GitHub workflow is included that will automatically build and deploy documentation to GitHub pages, after successful Release workflow (requires review and discussion with @jquense)

@jquense There is an issue with testing, after some of the dependency upgrades, that appear directly related to the `babel-preset-jason` preset in conjunction with Babel and Jest.

Submitting this now for review/help/feedback. 

**Note:** This does require a `yarn install` for fresh dependencies

ziga-miklic commented:
Should the new docs include info about the [utility functions](https://github.com/jquense/react-big-calendar/blob/master/src/utils/dates.js)? For example, we found the `firstVisibleDay()` and `lastVisibleDay()` useful in our project to get the current date range.

I also did a quick StackOverflow / GitHub search and it appears that others are using them as well.

cutterbl commented:
@ziga-miklic With all date math now being handled directly from the localizer, it might be a good idea to write some extensive localizer documentation. Good suggestion, I'll take a look at that.

cutterbl commented:
@jquense Have you had an opportunity to look at the tooling issues mentioned here?

cutterbl commented:
@ziga-miklic I added a `Localizers` guide to this documentation. Take a look and tell me if you think that's enough.

cutterbl commented:
@jquense I fixed the tooling. Testing functional. Added Contribution Guidelines, adjusted Issue, Feature Request, and Pull Request templates, plus minor workflow bits. Only thing to cover prior to merge is coordination on delivery directory for the GitHub pages auto build/deploy.

ziga-miklic commented on stories/guides/localizer.stories.mdx:
```diff
@@ -0,0 +1,50 @@
+
+# What is a Localizer?
+
+You have probably noticed that your Big Calendar implementations require a <LinkTo kind="props" story="localizer">localizer</LinkTo> prop. The `localizer` is needed for applying formatting and culture (i18n) to your date displays throughought the Calendar.
```
Is "throughought" a typo of "throughout"?

ziga-miklic commented on stories/guides/localizer.stories.mdx:
```diff
@@ -0,0 +1,50 @@
+- `firstVisibleDay(date:Date) => Date`
+- `format(value:Date, format:string) => string`,
+- `formats:object`
+- `getDstOffeset(start:Date, end:Date) => number`
```
Is `getDstOffeset` a typo of `getDstOffset`?

ziga-miklic commented on stories/guides/CustomView.stories.mdx:
````diff
@@ -0,0 +1,116 @@
+First build your custom `view` component. This could be based off an existing component, or something completely custom.
+
+```js
+import React from 'react'
````
Would it make sense to also import `useMemo` as it's used in the example?

ziga-miklic commented on stories/guides/CustomView.stories.mdx:
````diff
@@ -0,0 +1,116 @@
+```js
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Navigate as navigate } from 'react-big-calendar'
````
`Navigate` is renamed to `navigate` but in the example it's used as `Navigate` (`Navigate.NEXT` and `Navigate.PREVIOUS`). Do you think the renaming can be removed?

ziga-miklic commented on stories/guides/CustomView.stories.mdx:
```diff
@@ -0,0 +1,116 @@
+}
+
+CustomWeekView.title = (date, { localizer }) => {
+  const [start, ...rest] = Week.range(date, { localizer })
```
I think `Week` needs to be renamed to `CustomWeekView`. `Week` is not defined and I think we are trying to use the `range` method of our custom component?

ziga-miklic commented on stories/guides/CustomView.stories.mdx:
```diff
@@ -0,0 +1,116 @@
+
+CustomWeekView.range = (date, { localizer }) => {
+  const start = date
+  const end = dates.add(start, 2, 'day')
```
`dates` is not defined. I think we are trying to use the `add` localizer utility method. I think `dates` needs to be changed to `localizer`?

ziga-miklic commented:
@cutterbl Thank you for adding the docs for the Localizer utility methods. 👏 This would give us all the info we needed for our past use case.

I left a few comments of things that I noticed when reading the docs. Please take a look.

I also noticed that the "Run examples locally" section in README.md may need to be updated as it's still referencing the `yarn examples` command that has been removed in this PR.

cutterbl commented:
@ziga-miklic Nice catches on all of those. Late night/early morning fat fingering. Good to have second set of eyes. Thanks for looking out.

jquense commented:
@cutterbl there is really too much for me to look at here, i'm sure it's great. this is effectively your repro now, so don't wait on me for anything 👍

cutterbl commented:
@jquense Appreciate the vote of confidence there. As the owner though, only you can control the settings for the GitHub Pages, which is the final item in question here.

jquense commented:
what specifically do you need me to check or toggle? happy to do it

cutterbl commented:
@jquense just need to know what folder is setup with the 'gh-pages' branch in your Pages settings.

jquense commented:
it is root 👍  let me know if you want it to be something else

cutterbl commented:
@jquense New docs are live

jquense commented:
they look great! awesome work 👍

dpd73 commented:
Great work people. Dates/Timezone stuff particularly useful.

FYI, small typo on https://jquense.github.io/react-big-calendar/examples/index.html?path=/story/guides-understanding-dates--page

> ...culture, regarless of any offsets.

should be "regardless"
